app/src/main/java/com/skyyo/samples/features/signatureView/SignatureView.kt

```diff
@@ -0,0 +1,115 @@
+import androidx.compose.ui.input.pointer.pointerInteropFilter
+import androidx.compose.ui.layout.boundsInRoot
+import androidx.compose.ui.layout.onGloballyPositioned
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlin.math.roundToInt
+
+data class MotionEventValue(val eventType: Int, val x: Float, val y: Float)
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+fun SignatureView(
+    modifier: Modifier = Modifier,
+    stroke: Stroke = Stroke(10f),
+    paintColor: Color = Color.Black,
+    events: Flow<SignatureViewEvent>,
+    onBitmapSaved: (bitmap: Bitmap) -> Unit
+) {
+    val viewBounds = remember { mutableStateOf(Rect.Zero) }
+    val paint = remember {
+        Paint().apply {
+            style = PaintingStyle.Stroke
+            color = paintColor
+            strokeWidth = stroke.width
+        }
+    }
+    val savedList = rememberSaveable { mutableListOf<MotionEventValue>() }
+    val motionEventValue = remember { mutableStateOf<MotionEventValue?>(null) }
+
+    fun getBitmap(): Bitmap {
+        val bounds = viewBounds.value
+        val bitmap = Bitmap.createBitmap(
+            bounds.width.roundToInt(), bounds.height.roundToInt(),
+            Bitmap.Config.ARGB_8888
+        )
+        val canvasForSnapshot = Canvas(bitmap.asImageBitmap())
+        canvasForSnapshot.drawPath(savedList.toPath(), paint)
+
+        return bitmap
+    }
+
+    LaunchedEffect(Unit) {
+        events.collect { event ->
+            when (event) {
+                SignatureViewEvent.Reset -> {
+                    motionEventValue.value = null
+                    savedList.clear()
+                }
+                SignatureViewEvent.Save -> {
+                    val bitmap = getBitmap()
+                    onBitmapSaved(bitmap)
+                }
+            }
+        }
+    }
+
+    Canvas(
+        modifier = modifier
+            .onGloballyPositioned { viewBounds.value = it.boundsInRoot() }
+            .clipToBounds()
+            .pointerInteropFilter {
+                val x = it.x
+                val y = it.y
+                val value: MotionEventValue
+                when (it.action) {
+                    MotionEvent.ACTION_DOWN -> {
+                        value = MotionEventValue(it.action, x,y)
+                        motionEventValue.value = value
+                    }
+                    MotionEvent.ACTION_MOVE -> {
+                        value = MotionEventValue(it.action, x,y)
+                        motionEventValue.value = value
+                    }
+                }
+                true
+            },
+    ) {
+        motionEventValue.value.run {
+            this?.let {
+                savedList.add(it)
+            }
+            drawPath(
+                path = savedList.toPath(),
+                color = paintColor,
+                alpha = 1f,
+                style = Stroke(4f)
+            )
+        }
```

Comment:
1) Stroke can be remembered and reused
2) it's very inefficient to redraw every path on each `draw` call.Ideally we want to redraw all list only after PD. Otherwise - draw only what's new, canvas should persist the lines.
let's add a note to readme TODO:

- enhance signature view sample